kv/kvserver: TestReplicateRogueRemovedNode failed

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2657499&tab=buildLog) on [master@7853fd32de8b6dea869f2a2a92dcd7506f4a8998](https://github.com/cockroachdb/cockroach/commits/7853fd32de8b6dea869f2a2a92dcd7506f4a8998):

Fatal error:
```
panic: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r36: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>

```

Stack:
```
goroutine 341335 [running]:
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode.func2(0xc0027b5470, 0xd, 0x10, 0xc0073edd60, 0xc0073edd70, 0xc00468c000, 0xc00e16ad80, 0x300000003, 0x3, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3338 +0x3a5
created by github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3322 +0xea5

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode993083353
    test_log_scope.go:74: use -show-logs to present logs inline

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2666293&tab=buildLog) on [master@88c1ebeac2ac912ccada1e43ef212683151a2efe](https://github.com/cockroachdb/cockroach/commits/88c1ebeac2ac912ccada1e43ef212683151a2efe):

Fatal error:
```
panic: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>

```

Stack:
```
goroutine 346573 [running]:
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode.func2(0xc0083c42c0, 0xd, 0x10, 0xc00c4c3c10, 0xc00c4c3c20, 0xc0035c0d80, 0xc008dce180, 0x300000003, 0x3, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3338 +0x3a5
created by github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3322 +0xea5

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode928463409
    test_log_scope.go:74: use -show-logs to present logs inline

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2670474&tab=buildLog) on [master@e9e372122a2e3db7090b5705da07128f828e2441](https://github.com/cockroachdb/cockroach/commits/e9e372122a2e3db7090b5705da07128f828e2441):

Fatal error:
```
panic: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>

```

Stack:
```
goroutine 344573 [running]:
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode.func2(0xc008ea05c0, 0xd, 0x10, 0xc007f926d0, 0xc007f92710, 0xc003730000, 0xc008c3b380, 0x300000003, 0x3, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3338 +0x3a5
created by github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3322 +0xea5

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode586767866
    test_log_scope.go:74: use -show-logs to present logs inline

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2673770&tab=buildLog) on [master@9969862335cf501c1aa67a4aaf018ad5dc8e85e8](https://github.com/cockroachdb/cockroach/commits/9969862335cf501c1aa67a4aaf018ad5dc8e85e8):

Fatal error:
```
panic: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>

```

Stack:
```
goroutine 347235 [running]:
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode.func2(0xc0021871e0, 0xd, 0x10, 0xc006aa9c20, 0xc006aa9c30, 0xc005940480, 0xc009e98f00, 0x300000003, 0x3, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3338 +0x3a5
created by github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3322 +0xea5

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode389121155
    test_log_scope.go:74: use -show-logs to present logs inline

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2675029&tab=buildLog) on [master@9969862335cf501c1aa67a4aaf018ad5dc8e85e8](https://github.com/cockroachdb/cockroach/commits/9969862335cf501c1aa67a4aaf018ad5dc8e85e8):

Fatal error:
```
panic: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>

```

Stack:
```
goroutine 346770 [running]:
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode.func2(0xc010c47f10, 0xd, 0x10, 0xc00ef3b600, 0xc00ef3b610, 0xc002908000, 0xc003c0f980, 0x300000003, 0x3, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3338 +0x3a5
created by github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3322 +0xea5

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode431217540
    test_log_scope.go:74: use -show-logs to present logs inline

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2677001&tab=buildLog) on [master@ac0f819feeee435af2545034bbd40113b47d5b86](https://github.com/cockroachdb/cockroach/commits/ac0f819feeee435af2545034bbd40113b47d5b86):

Fatal error:
```
panic: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>

```

Stack:
```
goroutine 347913 [running]:
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode.func2(0xc009e348e0, 0xd, 0x10, 0xc00d129b20, 0xc00d129b30, 0xc00322e000, 0xc007df4480, 0x300000003, 0x3, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3338 +0x3a5
created by github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3322 +0xea5

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode009150343
    test_log_scope.go:74: use -show-logs to present logs inline

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2679534&tab=buildLog) on [master@68b86d57c7f36b9b6926d1fb82fda0cb5674a8f1](https://github.com/cockroachdb/cockroach/commits/68b86d57c7f36b9b6926d1fb82fda0cb5674a8f1):

Fatal error:
```
panic: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>

```

Stack:
```
goroutine 345628 [running]:
github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode.func2(0xc0092e20d0, 0xd, 0x10, 0xc0086aacd0, 0xc0086aace0, 0xc007214000, 0xc008dd6180, 0x300000003, 0x3, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3338 +0x3a5
created by github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateRogueRemovedNode
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_raft_test.go:3322 +0xea5

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode663413374
    test_log_scope.go:74: use -show-logs to present logs inline

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2684063&tab=buildLog) on [master@8a9268122caaf36ce80be35669388767393ea7a3](https://github.com/cockroachdb/cockroach/commits/8a9268122caaf36ce80be35669388767393ea7a3):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode249697014
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3336: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3385: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode249697014
--- FAIL: TestReplicateRogueRemovedNode (0.76s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2688809&tab=buildLog) on [master@86a44f733238b8999d1abe316aa48d3a3ccf5fc3](https://github.com/cockroachdb/cockroach/commits/86a44f733238b8999d1abe316aa48d3a3ccf5fc3):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode994027213
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3336: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3385: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode994027213
--- FAIL: TestReplicateRogueRemovedNode (0.69s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2693606&tab=buildLog) on [master@83e70ce84b740e27e721c3b73c38a4b8b515094a](https://github.com/cockroachdb/cockroach/commits/83e70ce84b740e27e721c3b73c38a4b8b515094a):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode350511387
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3336: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3385: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode350511387
--- FAIL: TestReplicateRogueRemovedNode (0.70s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2699731&tab=buildLog) on [master@64c4aef909f4382523cd9248341ca9f4448d841a](https://github.com/cockroachdb/cockroach/commits/64c4aef909f4382523cd9248341ca9f4448d841a):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode196002543
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3336: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3385: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode196002543
--- FAIL: TestReplicateRogueRemovedNode (0.71s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2708245&tab=buildLog) on [master@43174600d4e72030e9d71792c09935118f6e26ff](https://github.com/cockroachdb/cockroach/commits/43174600d4e72030e9d71792c09935118f6e26ff):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode603396226
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3396: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode603396226
--- FAIL: TestReplicateRogueRemovedNode (0.80s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2717208&tab=buildLog) on [master@26d86b709914e9a31ef92cfbf20d9c76ff8c149b](https://github.com/cockroachdb/cockroach/commits/26d86b709914e9a31ef92cfbf20d9c76ff8c149b):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode509999881
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3396: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode509999881
--- FAIL: TestReplicateRogueRemovedNode (0.74s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2721772&tab=buildLog) on [master@6601d827b814d4e85a1081b03bf2562d8ac2a4ab](https://github.com/cockroachdb/cockroach/commits/6601d827b814d4e85a1081b03bf2562d8ac2a4ab):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode323639287
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3396: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode323639287
--- FAIL: TestReplicateRogueRemovedNode (0.75s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2725499&tab=buildLog) on [master@dcc8ab04251aa954f086cd2b347cd1d1b976a312](https://github.com/cockroachdb/cockroach/commits/dcc8ab04251aa954f086cd2b347cd1d1b976a312):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode230558105
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3396: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode230558105
--- FAIL: TestReplicateRogueRemovedNode (0.72s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2726983&tab=buildLog) on [master@7d1324fa42732f482329a524b0166db8dd7365e6](https://github.com/cockroachdb/cockroach/commits/7d1324fa42732f482329a524b0166db8dd7365e6):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode291708172
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3396: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode291708172
--- FAIL: TestReplicateRogueRemovedNode (0.72s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2729135&tab=buildLog) on [master@206e8178de9ac124bfde5529d8a9b858734038bb](https://github.com/cockroachdb/cockroach/commits/206e8178de9ac124bfde5529d8a9b858734038bb):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode676528474
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    panic.go:617: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode676528474
--- FAIL: TestReplicateRogueRemovedNode (0.68s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2734063&tab=buildLog) on [master@1b18fc1fa8d9194d5e7f9aaa857425d80f0e4970](https://github.com/cockroachdb/cockroach/commits/1b18fc1fa8d9194d5e7f9aaa857425d80f0e4970):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode274703177
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3396: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode274703177
--- FAIL: TestReplicateRogueRemovedNode (0.73s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2739201&tab=buildLog) on [master@6de4313ec216161c79fe725fcc31fc87ef1804ea](https://github.com/cockroachdb/cockroach/commits/6de4313ec216161c79fe725fcc31fc87ef1804ea):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode215610136
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3396: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode215610136
--- FAIL: TestReplicateRogueRemovedNode (0.74s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2743539&tab=buildLog) on [master@37cdf01299861235888af3411e699480f5a2d670](https://github.com/cockroachdb/cockroach/commits/37cdf01299861235888af3411e699480f5a2d670):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode722051950
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3396: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode722051950
--- FAIL: TestReplicateRogueRemovedNode (0.76s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2747707&tab=buildLog) on [master@b703e663da8ededaee2e28fc39a24e3880ae54cf](https://github.com/cockroachdb/cockroach/commits/b703e663da8ededaee2e28fc39a24e3880ae54cf):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode131652846
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3396: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode131652846
--- FAIL: TestReplicateRogueRemovedNode (1.48s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2751746&tab=buildLog) on [master@15a185606d5e80b47d9fdd0ed4f54cfe29c527c6](https://github.com/cockroachdb/cockroach/commits/15a185606d5e80b47d9fdd0ed4f54cfe29c527c6):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode748130322
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3396: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode748130322
--- FAIL: TestReplicateRogueRemovedNode (1.49s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2753376&tab=buildLog) on [master@a69e6549a71f5a0e83eb13509001f4d7351050fb](https://github.com/cockroachdb/cockroach/commits/a69e6549a71f5a0e83eb13509001f4d7351050fb):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode156965971
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3347: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3396: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode156965971
--- FAIL: TestReplicateRogueRemovedNode (0.75s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2757791&tab=buildLog) on [master@93c38e97ddb9d18721eed850180c103fe3c45547](https://github.com/cockroachdb/cockroach/commits/93c38e97ddb9d18721eed850180c103fe3c45547):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode550138113
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3317: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3366: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode550138113
--- FAIL: TestReplicateRogueRemovedNode (0.75s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2762173&tab=buildLog) on [master@3bae381e45397225ff926f3f8f60d52b70fdc661](https://github.com/cockroachdb/cockroach/commits/3bae381e45397225ff926f3f8f60d52b70fdc661):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode926942211
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3317: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3366: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode926942211
--- FAIL: TestReplicateRogueRemovedNode (1.48s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2766077&tab=buildLog) on [master@fc68d927f946103a757888fcbd3ac20904c57a6b](https://github.com/cockroachdb/cockroach/commits/fc68d927f946103a757888fcbd3ac20904c57a6b):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode485544541
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3317: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3366: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode485544541
--- FAIL: TestReplicateRogueRemovedNode (0.75s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2774079&tab=buildLog) on [master@4d44ddf24153d8ef8e0a996fdbe75ac5607f9574](https://github.com/cockroachdb/cockroach/commits/4d44ddf24153d8ef8e0a996fdbe75ac5607f9574):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode191015627
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3317: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3366: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode191015627
--- FAIL: TestReplicateRogueRemovedNode (0.78s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicateRogueRemovedNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2775525&tab=buildLog) on [master@bdff5338ca725bf1cfddf7e3f648bbf02ab42999](https://github.com/cockroachdb/cockroach/commits/bdff5338ca725bf1cfddf7e3f648bbf02ab42999):

```
=== RUN   TestReplicateRogueRemovedNode
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode861358923
    test_log_scope.go:74: use -show-logs to present logs inline
    client_raft_test.go:3317: unexpected error: [NotLeaseHolderError] refusing to acquire lease on follower; r37: replica (n3,s3):3 not lease holder; current lease is repl=(n1,s1):1 seq=0 start=0,0 exp=<nil>
    client_raft_test.go:3366: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateRogueRemovedNode861358923
--- FAIL: TestReplicateRogueRemovedNode (1.48s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestReplicateRogueRemovedNode PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicateRogueRemovedNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/BUILD.bazel
+++ b/pkg/kv/kvserver/BUILD.bazel
@@ -285,7 +285,6 @@ go_test(
     data = glob(["testdata/**"]),
     embed = [":kvserver"],
     shard_count = 16,
-    tags = ["flaky"],
     deps = [
         "//pkg/base",
         "//pkg/cli/exit",

--- a/pkg/kv/kvserver/client_raft_test.go
+++ b/pkg/kv/kvserver/client_raft_test.go
@@ -3300,7 +3300,7 @@ func TestReplicateRogueRemovedNode(t *testing.T) {
 	replicaDesc, ok := rep.Desc().GetReplicaDescriptor(tc.Target(2).StoreID)
 	require.True(t, ok)
 
-	go func() {
+	require.NoError(t, tc.Stopper().RunAsyncTask(ctx, "send-req", func(ctx context.Context) {
 		incArgs := incrementArgs(key, 23)
 		startWG.Done()
 		defer finishWG.Done()
@@ -3312,11 +3312,18 @@ func TestReplicateRogueRemovedNode(t *testing.T) {
 				Timestamp: tc.Servers[2].Clock().Now(),
 			}, incArgs,
 		)
-		if _, ok := pErr.GetDetail().(*roachpb.RangeNotFoundError); !ok {
-			// We're on a goroutine, so cannot fatal.
+		detail := pErr.GetDetail()
+		switch detail.(type) {
+		case *roachpb.RangeNotFoundError:
+			// The typical case.
+		case *roachpb.NotLeaseHolderError:
+			// The atypical case - the lease may have expired and the
+			// lease acquisition may be refused.
+		default:
+			// NB: cannot fatal on a goroutine.
 			t.Errorf("unexpected error: %v", pErr)
 		}
-	}()
+	}))
 	startWG.Wait()
 	if t.Failed() {
 		t.FailNow()

